public/modules/custom/grants_application/src/User/GrantsProfile.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\grants_application\User;
+
+/**
+ * The grants profile class.
+ */
+class GrantsProfile {
+
+  /**
+   * The constructor.
+   *
+   * @param array $grantsProfileData
+   *   The raw data returned from outside system.
+   */
+  public function __construct(private array $grantsProfileData) {
+  }
+
+  /**
+   * Get the raw profile data array.
+   *
+   * CompanyShortName:string,
+   * companyName:string,
+   * companyHome:string,
+   * companyHomePage:string,
+   * companyStatus:string,
+   * companyStatusSpecial:string,
+   * businessPurpose:string,
+   * foundingYear:string,
+   * registrationDate:string
+   * officials:array,
+   * addresses:array,
+   * bankAccounts: array,
+   * businessId:string.
+   *
+   * @return array
+   *   The grants profile array.
+   */
+  public function toArray(): array {
+    return $this->grantsProfileData;
+  }
+
+  /**
+   * Find official from grants profile data by UUID.
+   *
+   * Array with keys:
+   * official_id: string,
+   * name: string,
+   * role: number,
+   * email: string,
+   * phone: string.
+   *
+   * @param string $uuid
+   *   The uuid.
+   *
+   * @return array
+   *   The official-array.
+   */
+  public function getCommunityOfficialByUuid(string $uuid): array {
```

Comment:
Applications can be sent without official, this should prolly account for undefined `$uuid`